### ISSUE TITLE
Compose file: Volume 'external' attribute is not being parsed correctly since v2.2.0

### ISSUE THREAD

**Issue by KloudJack** [docker/compose#8992]
Since the release of docker compose v2.2.0 the `external: true` compose file volume attribute doesn't seem to get parsed correctly.

**Steps to reproduce the issue:**

1. Create a very simple compose file which makes use of an external volume (see *additional information*)
2. Create a standard docker volume:
    `docker volume create my-postgres-db_data `
3. Attempt to bring up a container that makes use of said external volume with docker compose: 
    `docker compose up`

**Describe the results you received:**

Docker compose refuses to use the existing volume, even though it's defined with `external: true` as per the used Compose file spec for version 3:
```
volume "my-postgres-db_data" already exists but was not created by Docker Compose. Use `external: true` to use an existing volume
```

**Describe the results you expected:**

Docker compose should bring up the container with the external volume and not throw the above error.

Reverting to docker compose version 2.1.1 and attempting the reproduction steps results in the expected behaviour.

**Additional information you deem important (e.g. issue happens only occasionally):**

Example compose file for reproduction:
```
version: '3.9'

services:
  postgres:
    container_name: my-postgres-db
    image: postgres:13
    volumes:
      - db:/var/lib/postgresql/data

volumes:
  db:
    name: my-postgres-db_data
    external: true
```

**Output of `docker compose version`:**

```
Docker Compose version 2.2.0
```

**Output of `docker info`:**

```
Client:
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Build with BuildKit (Docker Inc., v0.6.1-docker)
  compose: Docker Compose (Docker Inc., 2.2.0)

Server:
 Containers: 3
  Running: 3
  Paused: 0
  Stopped: 0
 Images: 116
 Server Version: 20.10.11
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: false
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: systemd
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 io.containerd.runtime.v1.linux runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 1e5ef943eb76627a6d3b6de8cd1ef6537f393a71.m
 runc version: v1.0.2-0-g52b36a2d
 init version: de40ad0
 Security Options:
  seccomp
   Profile: default
  cgroupns
 Kernel Version: 5.10.83-1-lts
 Operating System: Arch Linux
 OSType: linux
 Architecture: x86_64
 CPUs: 4
 Total Memory: 15.59GiB
 Name: [ Hostname omitted ]
 ID: [ Host ID omitted ]
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
```

**Comment by KloudJack** [comment]
The issue seems to be resolved in docker compose v2.2.1+, I believe it's been fixed by this commit: bac732837e52cdbb6ba2a2176c448467bc4f621c

**Comment by kartas39** [comment]
issue still exists on compose version 2.2.2 aarch64
